File: 1111-maximum-nesting-depth-of-two-valid-parentheses-strings/1111-maximum-nesting-depth-of-two-valid-parentheses-strings.cpp
class Solution {
public:
    vector<int> maxDepthAfterSplit(string seq) {
        int n = seq.size();
        vector<int> ans(n);
        for (int i = 0, x = 0; i < n; ++i) {
            if (seq[i] == '(') {
                cout<<"before x: "<<x<<" || ";
                ans[i] = x++ & 1;
                cout<<"after x: "<<x<<" || ";
                cout<<"ansi: "<<ans[i]<<endl;
            } else {
                cout<<" ====== before x: "<<x<<" || ";
                ans[i] = --x & 1;
                cout<<" ====== after x: "<<x<<" || ";
                cout<<" ====== ansi: "<<ans[i]<<endl;
            }
        }
        return ans;
    }
};

File: 1111-maximum-nesting-depth-of-two-valid-parentheses-strings/1111-maximum-nesting-depth-of-two-valid-parentheses-strings.cs
public class Solution {
    public int[] MaxDepthAfterSplit(string seq) {
        int n = seq.Length;
        int[] ans = new int[n];
        for(int i=0, x=0; i < n; ++i) {
            if(seq[i] == '(') ans[i] = ++x & 1;
            else ans[i] = x-- & 1;
        }
        return ans;
    }
}